### PR TITLE
s22-6pm-2 Yufei include a STUB method in Add a Course API

### LINKED ISSUE #36: The add course API can return a customized error when course not found. Use a STUB method.

### PR DESCRIPTION
**Pull request by Sean-syf**
Create a stub method
The stub method returns true if enrollcode length is 5and quarter length is 5. Otherwise, return false.
It will be replaced by a course search method in #9 

If the stub method returns false then add course API throws a customized error and a message "course not found" followed by enrollCd and quarter

Screen Shot 2022-05-30 at 1 08 15 PM

Closes #36 

Implemented in #68

### CODE REVIEW

**Review by austinmacmath — DISMISSED** [review]
lgtm

**Review by austinmacmath — APPROVED** [review]
LGTM

**Review by mikeycurtis — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM... I like that we are doing a simple validation first, and then replacing it later with a better validation.  This is a nice incremental approach!

I'll test it on QA before merging.

File: src/main/java/edu/ucsb/cs156/courses/controllers/PersonalCoursesController.java [hunk 1]
```diff
@@ -71,8 +85,10 @@ public PersonalCourses addCourse(
             @ApiParam("enrollCd") @RequestParam String enrollCd,
             @ApiParam("psID") @RequestParam long psId,
             @ApiParam("Quarter (in yyyyq)") @RequestParam String quarter) {
-        CurrentUser currentUser = getCurrentUser();
-        log.info("currentUser={}", currentUser);
+        //FIX (Replace with course search method)
+        if(!STUBfunc(enrollCd,quarter)){
+            throw new EnrollCdInvalidException(enrollCd,quarter);
+        }
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/controllers/PersonalCoursesController.java:91]: This is good! 

But you should also validate the psId.

Do a personalScheduleRepository.find(psId) and if it doesn't find it, you should ALSO reject that as an invalid request.  You could throw an EntityNotFoundException.

**Review by pconrad — CHANGES_REQUESTED** [review]

**Pull request closed without merging**